solidity/dashboard/src/components/Modal.jsx

```diff
@@ -2,60 +2,81 @@ import React, { useEffect, useRef, useCallback, useState } from "react"
 import ReactDOM from "react-dom"
 import * as Icons from "./Icons"
 import ConfirmationModal from "./ConfirmationModal"
+import useCurrentWidthLowerThanThreshold from "../hooks/useCurrentWidthLowerThanThreshold"
+import MobileUsersModal from "./MobileUsersModal"
 
 const modalRoot = document.getElementById("modal-root")
 const crossIconHeight = 15
 const crossIconWidth = 15
 
-const Modal = React.memo(({ isOpen, closeModal, isFullScreen, ...props }) => {
-  const modalOverlay = useRef(null)
-  useEffect(() => {
-    if (isOpen) {
-      document.body.style.overflow = "hidden"
-    }
-
-    return () => {
-      document.body.style.overflow = "scroll"
-    }
-  }, [isOpen])
+const Modal = React.memo(
+  ({
+    isOpen,
+    closeModal,
+    isFullScreen,
+    hideTitleBar,
+    isRounded = false,
+    ...props
+  }) => {
+    const modalOverlay = useRef(null)
+    useEffect(() => {
+      if (isOpen) {
+        document.body.style.overflow = "hidden"
+      }
 
-  const onOverlayClick = useCallback(
-    (event) => {
-      if (modalOverlay.current === event.target) {
-        closeModal()
+      return () => {
+        document.body.style.overflow = "scroll"
       }
-    },
-    [closeModal]
-  )
+    }, [isOpen])
 
-  return isOpen
-    ? ReactDOM.createPortal(
-        <div
-          ref={modalOverlay}
-          className="modal-overlay"
-          onClick={onOverlayClick}
-        >
+    const onOverlayClick = useCallback(
+      (event) => {
+        if (modalOverlay.current === event.target) {
+          closeModal()
+        }
+      },
+      [closeModal]
+    )
+
+    const modalWrapperBorderStyle = isRounded
```

Comment:
I think we should get rid of `isRounded` props because as I can see on Figma we are going to round corners in most of the components, so we should address it in a separate PR. For now, I suggest using a custom class name or leaving w/o rounded corners.